### PR TITLE
reduce allocations when adding files

### PR DESCRIPTION
**Pull request by matthyx**
## Overview
This is with full_path enabled for trace_open and trace_openat:
image

```
GOMEMLIMIT=250MiB
resources:
  limits:
    cpu: 1500m
    memory: 600Mi
  requests:
    cpu: 500m
    memory: 600Mi                                                                                                                                                                
```

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -32,9 +32,11 @@ func LoadClusterData(path string) (ClusterData, error) {
...
 
 type Config struct {
-	EnableRelevancy  bool          `mapstructure:"relevantCVEServiceEnabled"`
-	MaxSniffingTime  time.Duration `mapstructure:"maxSniffingTimePerContainer"`
-	UpdateDataPeriod time.Duration `mapstructure:"updateDataPeriod"`
+	EnableFullPathTracing bool          `mapstructure:"fullPathTracingEnabled"`
+	EnableRelevancy       bool          `mapstructure:"relevantCVEServiceEnabled"`
+	InitialDelay          time.Duration `mapstructure:"initialDelay"`
+	MaxSniffingTime       time.Duration `mapstructure:"maxSniffingTimePerContainer"`
+	UpdateDataPeriod      time.Duration `mapstructure:"updateDataPeriod"`
```
Comment (dwertent) [inline pkg/config/config.go:39]: We should add this to the helm chart as well

Comment (dwertent) [reply pkg/config/config.go:39]: Also, we should have default values (if missing from config)

**Review by dwertent — APPROVED** [review]
Well done!
Please see my comments :)

File: pkg/config/config.go (hunk 1 above)
Comment (matthyx) [reply pkg/config/config.go:39]: like in line https://github.com/kubescape/node-agent/pull/110/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R48 ?

Comment (matthyx) [reply pkg/config/config.go:39]: what we should modify in the helm chart is the `updateDataPeriod` as well as the resources...

**Pull request merged**